### PR TITLE
lxd: add forkusernsexec()

### PR DESCRIPTION
**Pull request by brauner**
Use from LXD like:

```
args := []string{
	d.os.ExecPath,
	"forkusernsexec",
	"touch",
	"/tmp/MUHU",
}
cmd := exec.Cmd{}
cmd.Path = d.os.ExecPath
cmd.Args = args

rUidMapPipe, wUidMapPipe, err := os.Pipe()
if err != nil {
	return err
}
defer rUidMapPipe.Close()

// Make sure the string is \x00 terminated so it's parseable as a C string
_, err = wUidMapPipe.WriteString("0 100000 1\n\x00")
if err != nil {
	return err
}

rGidMapPipe, wGidMapPipe, err := os.Pipe()
if err != nil {
	return err
}
defer rGidMapPipe.Close()

// Make sure the string is \x00 terminated so it's parseable as a C string
_, err = wGidMapPipe.WriteString("0 100000 1\n\x00")
if err != nil {
	return err
}

cmd.ExtraFiles = []*os.File{rUidMapPipe, rGidMapPipe}
err = cmd.Run()
if err != nil {
	return err
}
```

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: lxd/main_forkusernsexec.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+import "C"
+
+type cmdForkusernsexec struct {
+	global *cmdGlobal
+}
+
+func (c *cmdForkusernsexec) Command() *cobra.Command {
+	// Main subcommand
+	cmd := &cobra.Command{}
+	cmd.Use = "forkusernsexec <cmd> <args>"
```
Comment (tomponline) [inline lxd/main_forkusernsexec.go:202]: @brauner @stgraber shouldn't this take 2 FDs for the uid and gid map files?

Comment (stgraber) [reply lxd/main_forkusernsexec.go:202]: They are at well known locations, it's expecting fd3 and fd4 to be set.

Comment (tomponline) [reply lxd/main_forkusernsexec.go:202]: Ah OK it would be good to document that in the command help for future users though.

Comment (brauner) [reply lxd/main_forkusernsexec.go:244]: This is simply a Go thing not something we do.

**Comment by tomponline** [comment]
thanks for this @brauner

**Comment by brauner** [comment]
Failure on Jenkins due to @stgraber's changes. Restarting tests now.

**Comment by brauner** [comment]
jenkins: test this please

**Pull request merged**